src/bentoml/models.py

```diff
@@ -227,6 +227,50 @@ def pull(
     return _cloud_client.pull_model(tag, force=force)
 
 
+if t.TYPE_CHECKING:
+
+    class CreateKwargs(t.TypedDict):
+        module: str
+        api_version: str | None
+        signatures: t.Required[ModelSignaturesType]
+        labels: dict[str, t.Any] | None
+        options: ModelOptions | None
+        custom_objects: dict[str, t.Any] | None
+        external_modules: t.List[ModuleType] | None
+        metadata: dict[str, t.Any] | None
+        context: t.Required[ModelContext]
+        use_tempfs: t.NotRequired[bool]
+        _model_store: t.NotRequired[ModelStore]
+
+
+@t.overload
+@contextmanager
+def create(
+    name: Tag | str, /, **attrs: t.Unpack[CreateKwargs]
+) -> t.Generator[Model, None, None]:
+    ...
+
+
+@t.overload
+@contextmanager
+def create(
+    name: Tag | str,
+    *,
+    module: str = ...,
+    api_version: str | None = ...,
+    signatures: ModelSignaturesType,
+    labels: dict[str, t.Any] | None = ...,
+    options: ModelOptions | None = ...,
+    custom_objects: dict[str, t.Any] | None = ...,
+    external_modules: t.List[ModuleType] | None = ...,
+    metadata: dict[str, t.Any] | None = ...,
+    context: ModelContext,
+    use_tempfs: bool = ...,
+    _model_store: ModelStore = ...,
+) -> t.Generator[Model, None, None]:
+    ...
+
+
```

Comment:
I'm a little bit lost here but what's purpose of writing definitions in this way?

Reply:
Oh this is to support create with unpack
```python
kwargs = dict(context=..., not_a_valid_type="something")
with bentoml.models.create("some_name", **kwargs): # <- should throw an error here saying kwargs doesn't have a valid argument
```